<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from man_control.xacro              | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="man_control">
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <robotNamespace>/manipulator</robotNamespace>
      <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
    </plugin>
  </gazebo>
  <link name="world"/>
  <link name="L0">
    <inertial>
      <origin rpy="0 0 0" xyz="-2.77555756156289E-17 1.38777878078145E-17 0.251308543489449"/>
      <mass value="8.58391105208198"/>
      <inertia ixx="0.521593432983479" ixy="0" ixz="4.10570392569217E-17" iyy="0.521593432983479" iyz="-1.43136405786465E-32" izz="0.0439830975047331"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://man_urdf/meshes/L0.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://man_urdf/meshes/L0.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="J0" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world"/>
    <child link="L0"/>
    <axis xyz="0 0 1"/>
    <limit effort="10000" lower="-3.1416" upper="3.1416" velocity="100"/>
  </joint>
  <link name="L1">
    <inertial>
      <origin rpy="0 0 0" xyz="2.83887343077716E-08 -0.0178750899362703 -0.159236806529472"/>
      <mass value="16.8313777816089"/>
      <inertia ixx="0.689772480581919" ixy="2.28727857677598E-08" ixz="-2.27577952071559E-07" iyy="0.631143643333626" iyz="0.0479077343963207" izz="0.156178494391246"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://man_urdf/meshes/L1.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://man_urdf/meshes/L1.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="J1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.55"/>
    <parent link="L0"/>
    <child link="L1"/>
    <axis xyz="0 0 1"/>
    <limit effort="10000" lower="-3.1416" upper="3.1416" velocity="100"/>
  </joint>
  <link name="L2">
    <inertial>
      <origin rpy="0 0 0" xyz="0.266754722157752 0 0.230046531880633"/>
      <mass value="16.370380678601"/>
      <inertia ixx="0.134812941774803" ixy="-2.66274557242285E-17" ixz="0.0761886064644828" iyy="0.734239754509835" iyz="1.40933669931598E-17" izz="0.724145816763749"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://man_urdf/meshes/L2.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.792156862745098 1 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://man_urdf/meshes/L2.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="J2" type="revolute">
    <origin rpy="1.5708 0 0" xyz="0 0 0"/>
    <parent link="L1"/>
    <child link="L2"/>
    <axis xyz="0 0 1"/>
    <limit effort="10000" lower="-1.5708" upper="1.5708" velocity="100"/>
  </joint>
  <link name="L3">
    <inertial>
      <origin rpy="0 0 0" xyz="0 -0.112376756749049 -4.16333634234434E-17"/>
      <mass value="8.41901264308844"/>
      <inertia ixx="0.154818236954621" ixy="-3.62937528884453E-17" ixz="-8.85398612101813E-19" iyy="0.0380822097268156" iyz="-1.93424736843363E-17" izz="0.179225930646112"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://man_urdf/meshes/L3.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.792156862745098 0.819607843137255 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://man_urdf/meshes/L3.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="J3" type="revolute">
    <origin rpy="0 0 1.5708" xyz="0.5 0 0"/>
    <parent link="L2"/>
    <child link="L3"/>
    <axis xyz="0 0 1"/>
    <limit effort="10000" lower="-1.5708" upper="1.5708" velocity="100"/>
  </joint>
  <link name="L4">
    <inertial>
      <origin rpy="0 0 0" xyz="-3.49246999764574E-07 -1.92054451633727E-07 -0.0344320239635494"/>
      <mass value="4.31368808813062"/>
      <inertia ixx="0.046306035676263" ixy="1.33899030712816E-07" ixz="2.46671103312782E-08" iyy="0.0356858333371746" iyz="-1.46848990830211E-07" izz="0.0304029387728525"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://man_urdf/meshes/L4.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.5 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://man_urdf/meshes/L4.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="J4" type="revolute">
    <origin rpy="1.5708 0 0" xyz="0 -0.5 0"/>
    <parent link="L3"/>
    <child link="L4"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="-3.1416" upper="3.1416" velocity="100"/>
  </joint>
  <link name="L5">
    <inertial>
      <origin rpy="0 0 0" xyz="6.83871559381544E-09 -0.0305734791822949 -3.70412332750814E-08"/>
      <mass value="0.793971101412226"/>
      <inertia ixx="0.00315651901404138" ixy="1.83991622804642E-10" ixz="2.39350805523045E-10" iyy="0.00096585390013851" iyz="-2.25178105054635E-09" izz="0.00315072678838859"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://man_urdf/meshes/L5.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.792156862745098 0.5 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://man_urdf/meshes/L5.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="J5" type="revolute">
    <origin rpy="-1.5708 0 0" xyz="0 0 0"/>
    <parent link="L4"/>
    <child link="L5"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="-1.5708" upper="1.5708" velocity="100"/>
  </joint>
  <link name="L6">
    <inertial>
      <origin rpy="0 0 0" xyz="0 8.32667268468867E-17 0.179502718586101"/>
      <mass value="0.260445526746461"/>
      <inertia ixx="0.000240259603757247" ixy="6.91705639679729E-20" ixz="4.87341293492895E-20" iyy="0.000240259603757247" iyz="2.23501304756425E-20" izz="0.000242604285420145"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://man_urdf/meshes/L6.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.792156862745098 0.819607843137255 0.5 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://man_urdf/meshes/L6.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="J6" type="continuous">
    <origin rpy="1.5708 0 0" xyz="0 0 0"/>
    <parent link="L5"/>
    <child link="L6"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" velocity="100"/>
  </joint>
  <gazebo reference="L0">
    <selfCollide>true</selfCollide>
    <collision>
      <surface>
        <bounce>
          <restitution_coefficient>0.2</restitution_coefficient>
        </bounce>
      </surface>
    </collision>
  </gazebo>
  <gazebo reference="L1">
    <selfCollide>true</selfCollide>
    <collision>
      <surface>
        <bounce>
          <restitution_coefficient>0.2</restitution_coefficient>
        </bounce>
      </surface>
    </collision>
  </gazebo>
  <gazebo reference="L2">
    <selfCollide>true</selfCollide>
    <collision>
      <surface>
        <bounce>
          <restitution_coefficient>0.2</restitution_coefficient>
        </bounce>
      </surface>
    </collision>
  </gazebo>
  <gazebo reference="L3">
    <selfCollide>true</selfCollide>
    <collision>
      <surface>
        <bounce>
          <restitution_coefficient>0.2</restitution_coefficient>
        </bounce>
      </surface>
    </collision>
  </gazebo>
  <gazebo reference="L4">
    <selfCollide>true</selfCollide>
    <collision>
      <surface>
        <bounce>
          <restitution_coefficient>0.2</restitution_coefficient>
        </bounce>
      </surface>
    </collision>
  </gazebo>
  <gazebo reference="L5">
    <selfCollide>true</selfCollide>
    <collision>
      <surface>
        <bounce>
          <restitution_coefficient>0.2</restitution_coefficient>
        </bounce>
      </surface>
    </collision>
  </gazebo>
  <gazebo reference="L6">
    <selfCollide>true</selfCollide>
    <collision>
      <surface>
        <bounce>
          <restitution_coefficient>0.2</restitution_coefficient>
        </bounce>
      </surface>
    </collision>
  </gazebo>
  <transmission name="J1_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="J1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="J1_motor">
      <mechanicalReduction>100</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="J2_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="J2">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="J2_motor">
      <mechanicalReduction>100</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="J3_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="J3">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="J3_motor">
      <mechanicalReduction>100</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="J4_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="J4">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="J4_motor">
      <mechanicalReduction>100</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="J5_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="J5">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="J5_motor">
      <mechanicalReduction>100</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="J6_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="J6">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="J6_motor">
      <mechanicalReduction>100</mechanicalReduction>
    </actuator>
  </transmission>
</robot>
